src/peergos/server/tests/simulation/Simulator.java

```diff
@@ -486,44 +558,79 @@ private boolean verifySharingPermissions(String user, Path path) {
-                    case WRITE:
-                        PeergosFileSystemImpl fs = fileSystems.getTestFileSystem(sharee);
-                        try {
-                            //can read?
-                            read = fs.read(path);
-                        } catch (Exception ex) {
-                            LOG.log(Level.SEVERE, "User "+ sharee +" could not read shared-path "+ path +"!", ex);
-                            isVerified = false;
-                        }
-                        if (isVerified) {
+                PeergosFileSystemImpl fs = fileSystems.getTestFileSystem(sharee);
+                if (isDir) {
+                    switch (permission) {
+                        case READ:
+                            try {
+                                //can read?
+                                fs.ls(path);
+                            } catch (Exception ex) {
+                                LOG.log(Level.SEVERE, "User " + sharee + " could not read shared-path " + path + "!", ex);
+                                isVerified = false;
+                            }
+                            break;
+                        case WRITE:
+                            try {
+                                //can read?
+                                fs.ls(path);
+                            } catch (Exception ex) {
+                                LOG.log(Level.SEVERE, "User " + sharee + " could not read shared-path " + path + "!", ex);
+                                isVerified = false;
+                            }
+                            if (isVerified) {
+                                try {
+                                    mkdir(user, path.resolve(getNextName()));
+                                } catch (Exception ex) {
+                                    LOG.log(Level.SEVERE, "User " + sharee + " could not write  shared-path " + path + "!", ex);
+                                    isVerified = false;
+                                }
+                            }
+                            break;
+                        default:
+                            throw new IllegalStateException();
+                    }
+                } else {
+                    switch (permission) {
+                        case READ:
+                            try {
+                                //can read?
+                                read = fs.read(path);
+                            } catch (Exception ex) {
+                                LOG.log(Level.SEVERE, "User " + sharee + " could not read shared-path " + path + "!", ex);
+                                isVerified = false;
+                            }
+                            break;
+                        case WRITE:
                             try {
-                                // can overwrite?
-                                fs.write(path, new byte[]{read[0]});
+                                //can read?
```

Comment:
Isn't this no longer checking that the path can be written? It's only being read now.